### PR TITLE
Fix clang build fail

### PR DESCRIPTION
**Pull request by yuhaoth**
@cielavenir, I re-org your commits . Please review that.  I think your patches have two topics

-  Clang support : It can be verified with current test matrix.
-  Arm64 Apple support: It can not be tested now. Please help review and verified it .

I am not sure if you think it is reasonable to merge this PR and abandon #162 and #164 ?
If No, Could you re-org your patches ? 

@gbtucker  , please review `include/aarch64_multibinary.h` . I am not sure if there are copy rights issues

### CODE REVIEW

**Comment by cielavenir** [comment]
@yuhaoth I think it is ok. Commit message is also fine.

**Comment by cielavenir** [comment]
@yuhaoth emailed about copy rights issues (and suggestions) cc-ing Greg-san. I'd like to avoid confusing users until the situation is settled.

File: include/aarch64_multibinary.h [hunk 1]
```diff
@@ -168,32 +182,65 @@
...
 .endm
 
 #else /* __ASSEMBLY__ */
+#include <stdint.h>
+#if defined(__linux__)
 #include <sys/auxv.h>
-
+#elif defined(__APPLE__)
+int _get_cpu_capabilities(void);
+#define kHasICDSBShift                  2
```
Comment (yuhaoth) [inline include/aarch64_multibinary.h:224]: This part looks same with the code in  https://opensource.apple.com/source/xnu/xnu-6153.81.5/osfmk/arm/cpu_capabilities.h.auto.html .
It is release under http://www.opensource.apple.com/apsl/ . 

I am not sure if it is conflict with BSD license . 
@gbtucker  please review .

**Comment by yuhaoth** [comment]
This PR pass at https://travis-ci.org/github/yuhaoth/isa-l/builds/748306963

latest update is to remove some local tag in commit message :)
@gtucker , please review

**Comment by cielavenir** [comment]
for now please WIP this PR.

**Comment by yuhaoth** [comment]
Apple support has been removed . 

@gbtucker  please review

**Comment by yuhaoth** [comment]
it is passed at https://travis-ci.org/github/yuhaoth/isa-l/builds/748495536

**Comment by gbtucker** [comment]
Thanks @yuhaoth and @cielavenir for working together and separating out the clang issue from other.  I can schedule this part for integration and leave the other as WIP for now.

**Pull request merged**